src/lib.rs

```diff
@@ -1305,26 +1316,29 @@ pub mod pallet {
 		///
 		/// - `disputes`: The votes for disputes, containing the dispute identifier and whether the caller supports or is against the vote. None indicates the caller’s classification of the dispute as invalid.
 		#[pallet::call_index(11)]
-		#[pallet::weight(343852000)]
+		#[pallet::weight(<T as Config>::WeightInfo::vote_on_multiple_disputes(T::MaxDisputeVotes::get()))]
 		pub fn vote_on_multiple_disputes(
 			origin: OriginFor<T>,
 			votes: BoundedVec<(DisputeId, Option<bool>), T::MaxDisputeVotes>,
-		) -> DispatchResult {
+		) -> DispatchResultWithPostInfo {
 			let voter = ensure_signed(origin)?;
-			for (dispute_id, supports) in votes {
-				Self::do_vote(&voter, dispute_id, supports)?
+			for (dispute_id, supports) in &votes {
+				Self::do_vote(&voter, *dispute_id, *supports)?
 			}
-			Ok(())
+			Ok(Some(T::WeightInfo::vote_on_multiple_disputes(votes.len() as u32)).into())
 		}
 
 		/// Sends any pending dispute votes due to the governance controller contract for tallying.
 		///
 		/// - `max_votes`: The maximum number of votes to be sent.
 		#[pallet::call_index(12)]
-		#[pallet::weight(1218085000)]
-		pub fn send_votes(origin: OriginFor<T>, max_votes: u8) -> DispatchResult {
+		#[pallet::weight(<T as Config>::WeightInfo::send_votes(u8::MAX.into()))]
+		pub fn send_votes(origin: OriginFor<T>, max_votes: u8) -> DispatchResultWithPostInfo {
 			ensure_signed(origin)?;
-			Self::do_send_votes(Self::now(), max_votes)
+			match Self::do_send_votes(Self::now(), max_votes) {
```

Comment:
Can this not be simplified via the `?` operator?

Something like `Ok(Some(T::WeightInfo::vote_on_multiple_disputes(vote_count)?).into())`

Reply:
Thanks for pointing out